addons/payment_paypal/tests/test_paypal.py

```diff
@@ -85,3 +85,12 @@ def test_webhook_notification_triggers_origin_check(self):
         ):
             self._make_json_request(url, data=self.notification_data)
             self.assertEqual(origin_check_mock.call_count, 1)
+
+    def test_processing_values_public_user(self):
+        """ If a payment is made from the Point of Sale the user is the public user. In this case we need to make sure that
+            the email address is not sent to PayPal and that we provide the country code of the company instead."""
+        self.partner = self.env.ref('base.public_user')
```

Comment:
It's not super clear that overwriting `self.partner` ensures that it's the one being assigned to the transaction's `partner_id`. Pass `partner_id` directly in the `_create_transaction` call instead.